streams/test-utils/src/test/java/org/apache/kafka/streams/test/ConsumerRecordFactoryConsumer.java

```diff
@@ -0,0 +1,289 @@
+            fail("Should have throw IllegalArgumentException");
+        } catch (final IllegalArgumentException expected) { /* pass */ }
+
+        try {
+            factory.create(nullTopicName, Collections.singletonList(KeyValue.pair(rawKey, value)), timestamp, 2L);
+            fail("Should have throw IllegalArgumentException");
+        } catch (final IllegalArgumentException expected) { /* pass */ }
+    }
+
+    @Test
+    public void shouldRequireCustomTopicNameIfNotDefaultFactoryTopicName() {
+        final ConsumerRecordFactory<byte[], Integer> factory =
+            new ConsumerRecordFactory<>(new ByteArraySerializer(), integerSerializer);
+
+        try {
+            factory.create(rawKey, value, timestamp);
+            fail("Should have throw IllegalStateException");
+        } catch (final IllegalStateException expected) { /* pass */ }
+
+        try {
+            factory.create(rawKey, value);
+            fail("Should have throw IllegalStateException");
+        } catch (final IllegalStateException expected) { /* pass */ }
+
+        try {
+            factory.create(value, timestamp);
+            fail("Should have throw IllegalStateException");
+        } catch (final IllegalStateException expected) { /* pass */ }
+
+        try {
+            factory.create(value);
+            fail("Should have throw IllegalStateException");
+        } catch (final IllegalStateException expected) { /* pass */ }
+
+        try {
+            factory.create(Collections.singletonList(KeyValue.pair(rawKey, value)));
+            fail("Should have throw IllegalStateException");
+        } catch (final IllegalStateException expected) { /* pass */ }
+
+        try {
+            factory.create(Collections.singletonList(KeyValue.pair(rawKey, value)), timestamp, 2L);
+            fail("Should have throw IllegalStateException");
+        } catch (final IllegalStateException expected) { /* pass */ }
+    }
+
+    @Test
+    public void shouldCreateConsumerRecordWithOtherTopicNameAndTimestamp() {
+        record = factory.create(otherTopicName, rawKey, value, timestamp);
+        verifyRecord(otherTopicName, rawKey, rawValue, timestamp, record);
+    }
+
+    @Test
+    public void shouldCreateConsumerRecordWithTimestamp() {
+        record = factory.create(rawKey, value, timestamp);
+        verifyRecord(topicName, rawKey, rawValue, timestamp, record);
+    }
+
+    @Test
+    public void shouldCreateConsumerRecordWithOtherTopicName() {
+        record = factory.create(otherTopicName, rawKey, value);
+        verifyRecord(otherTopicName, rawKey, rawValue, 0L, record);
+
+        factory.advanceTimeMs(3L);
+        record = factory.create(otherTopicName, rawKey, value);
+        verifyRecord(otherTopicName, rawKey, rawValue, 3L, record);
+    }
+
+    @Test
+    public void shouldCreateConsumerRecord() {
+        record = factory.create(rawKey, value);
```

Comment:
why save `record` to a field? Can just inline it in call to `verifyRecord` - elsewhere too